app/controllers/todos_controller.rb

```diff
@@ -47,19 +57,19 @@ def destroy
 
   # Set the TodoList by user_id and todo_list_id
   def set_todo_list
-    @todo_list = TodoList.find_by(id: params[:todo_list_id], user_id: params[:user_id])
-    unless @todo_list
-      logger.error "Todo list not found: #{params[:todo_list_id]}"
-      render json: { error: 'Todo list not found' }, status: :not_found
-    end
+    # Fetch the TodoList owned by the user or where the user is a collaborator
+    @todo_list = TodoList.joins("LEFT JOIN collaborators ON collaborators.todo_list_id = todo_lists.id")
+                         .where("todo_lists.user_id = :user_id OR collaborators.user_id = :user_id", user_id: current_user.id)
+                         .find_by(id: params[:todo_list_id])
```

Comment:
Can't we use policy scope here?

Reply:
Yes, we can use the policy scope here instead of manually joining tables and writing authorization logic in the controller.